bin/lock_dev_tool.ml

```diff
@@ -130,25 +125,37 @@ let locked_ocaml_compiler_version () =
             [ Pp.text "Try running"; User_message.command "dune pkg lock" ]
         ]
   | Ok { packages; _ } ->
-    let packages = Lock_dir.Packages.pkgs_on_platform_by_name packages ~platform in
-    (match Package_name.Map.find packages compiler_package_name with
-     | None ->
-       User_error.raise
-         [ Pp.textf
-             "The lockdir doesn't contain a lockfile for the package %S."
-             (Package_name.to_string compiler_package_name)
-         ]
-         ~hints:
-           [ Pp.concat
-               ~sep:Pp.space
-               [ Pp.textf
-                   "Add a dependency on %S to one of the packages in dune-project and \
-                    then run"
-                   (Package_name.to_string compiler_package_name)
-               ; User_message.command "dune pkg lock"
-               ]
-           ]
-     | Some pkg -> Memo.return pkg.info.version)
+    Memo.return (Lock_dir.Packages.pkgs_on_platform_by_name packages ~platform)
+;;
+
+let lookup_package_exn packages pkg_name =
+  match Package_name.Map.find packages pkg_name with
+  | Some p -> p
+  | None ->
+    User_error.raise
+      [ Pp.textf
+          "The lockdir doesn't contain a lockfile for the package %S."
+          (Package_name.to_string pkg_name)
+      ]
+      ~hints:
+        [ Pp.concat
+            ~sep:Pp.space
+            [ Pp.textf
+                "Add a dependency on %S to one of the packages in dune-project and then \
+                 run"
+                (Package_name.to_string pkg_name)
+            ; User_message.command "dune pkg lock"
```

Comment:
I would drop the recommendation to run `dune pkg lock` here. We probably don't want to suggest people that use autolocking to suddenly opt out of it.